File: backend-spring-boot/src/main/java/com/licenta/foodtrack/dto/MasaResponse.java
package com.licenta.foodtrack.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.LocalDate;
import java.util.List;

public record MasaResponse(
        Long id,
        @JsonProperty("categorie_masa_id")
        Long categorieMasaId,
        LocalDate data,
        String ora,
        String notite,

        @JsonProperty("energy_kcal_total")
        Double energyKcalTotal,

        @JsonProperty("energy_kj_total")
        Double energyKjTotal,

        @JsonProperty("fat_total")
        Double fatTotal,

        @JsonProperty("fat_percent")
        Double fatPercent,

        @JsonProperty("saturated_fat_total")
        Double saturatedFatTotal,

        @JsonProperty("carbohydrates_total")
        Double carbohydratesTotal,

        @JsonProperty("carbohydrates_percent")
        Double carbohydratesPercent,

        @JsonProperty("sugars_total")
        Double sugarsTotal,

        @JsonProperty("fiber_total")
        Double fiberTotal,

        @JsonProperty("protein_total")
        Double proteinTotal,

        @JsonProperty("protein_percent")
        Double proteinPercent,

        @JsonProperty("salt_total")
        Double saltTotal,

        List<InregistrareAlimentResponse> alimente
) {
}

File: frontend-flutter/frontend_flutter/lib/features/mese/models/masa_response.dart
import 'dart:ffi';

import 'inregistrare_aliment_response.dart';

class MasaResponse {
  final int id;
  final int? categorieMasaId;
  final String data;
  final String? ora;
  final String? notite;
  final double? gramsTotal;
  final double? energyKcalTotal;
  final double? fatTotal;
  final double? saturatedFatTotal;
  final double? carbohydratesTotal;
  final double? sugarsTotal;
  final double? fiberTotal;
  final double? proteinTotal;
  final double? saltTotal;
  final double? proteinPercent;
  final double? carbohydratesPercent;
  final double? fatPercent;
  final List<InregistrareAlimentResponse> alimente;

  MasaResponse({
    required this.id,
    this.categorieMasaId,
    required this.data,
    this.ora,
    this.notite,
    this.gramsTotal,
    this.energyKcalTotal,
    this.fatTotal,
    this.carbohydratesTotal,
    this.fiberTotal,
    this.saturatedFatTotal,
    this.proteinTotal,
    this.saltTotal,
    this.proteinPercent,
    this.carbohydratesPercent,
    this.sugarsTotal,
    this.fatPercent,
    required this.alimente,
  });

  factory MasaResponse.fromJson(Map<String, dynamic> json) {
    num? parseNum(dynamic v) =>
        v is num ? v : (v == null ? null : num.tryParse(v.toString()));

    final alimenteJson = (json['alimente'] as List<dynamic>?) ?? [];

    return MasaResponse(
      id: (json['id'] as num).toInt(),
      categorieMasaId: (json['categorie_masa_id'] as num?)?.toInt(),
      data: json['data'] as String? ?? '',
      ora: json['ora'] as String?,
      notite: json['notite'] as String?,
      gramsTotal: parseNum(json['grams_total'])?.toDouble(),
      energyKcalTotal: parseNum(json['energy_kcal_total'])?.toDouble(),
      fatTotal: parseNum(json['fat_total'])?.toDouble(),
      carbohydratesTotal: parseNum(json['carbohydrates_total'])?.toDouble(),
      saturatedFatTotal: parseNum(json['saturated_fat_total'])?.toDouble(),
      fiberTotal: parseNum(json['fiber_total'])?.toDouble(),
      sugarsTotal: parseNum(json['sugars_total'])?.toDouble(),
      proteinTotal: parseNum(json['protein_total'])?.toDouble(),
      saltTotal: parseNum(json['salt_total'])?.toDouble(),
      proteinPercent: parseNum(json['protein_percent'])?.toDouble(),
      carbohydratesPercent: parseNum(json['carbohydrates_percent'])?.toDouble(),
      fatPercent: parseNum(json['fat_percent'])?.toDouble(),
      alimente: alimenteJson
          .map((e) => InregistrareAlimentResponse.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }
}